### PR TITLE
Rails Capstone : Expense Controller App

### PR DESCRIPTION
**Pull request by patriciachrysy**
Hello dear TSE, thank you for taking time to review my app. This is what has bee done so far:
Splash screen

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
Home page (categories page)

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/expenses_manager/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+<div class="expenses">
+    <% if @expenses.count > 0 %>
+        <% @expenses.each do |expense| %>
+            <%= render 'expense', {category: expense.categories.first, expense: expense} %>
+        <% end %>
+    <% else %>
+        <small> No expense added .</small>
+    <% end %>
+</div>
```
Comment (thecodechaser) [inline app/views/expenses_manager/index.html.erb:14]: - [x] Please make sure the transactions are presented, ordered by the most recent as requested in the project requirements. Currently, they are ordered by the least recent.

image

File: app/views/expenses_manager/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<%= stylesheet_link_tag "application" %>
+<%= stylesheet_link_tag "expense" %>
+
+<%= render "layouts/nav_bar", {prev_path: nil, title: 'new expense', from_form: false, next_path: categories_path, next_text: 'search' } %>
+
+<div class="expense-form">
+    <%= render 'form', {category: @category, expense: @expense, submit_path: category_expenses_manager_index_path(@category, @expense)} %>
+</div>
```
Comment (thecodechaser) [inline app/views/expenses_manager/new.html.erb]: - [x] Please make sure on the `add new transaction` page when the user clicks on the "Back" button (<), the user is redirected to the transactions page for that category as requested in the project requirements. Currently, the user is redirected to the home page(categories page).

image

File: app/views/categories/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    <% if @categories.count > 0 %>
+        <% @categories.each do |category| %>
+            <%= render category %>
+        <% end %>
+    <% else %>
+        <small> No categories added, <%= link_to "click", new_category_path, class: 'clear-button' %> to add.</small>
+    <% end %>
+</div>
+
+<%= link_to raw("<i class='las la-plus'></i>"), new_category_path, class: 'primary-button floating-button button-round' if can? :create, Category %>
```
Comment (thecodechaser) [inline app/views/categories/index.html.erb]: - [x] Please make sure to use the below styles for the `add new category` and `add new transaction` buttons to follow the given design guidelines.

image

File: README.md [hunk 4]
```diff
@@ -103,7 +104,11 @@
 
 ### Deployment
 
-**This project is deployed by the author, no permission for deployment by any other client.**
+- Find the link to the deployed version here [Expense Controller](https://expense-controller-app.onrender.com/)
+
+### Presentation video
+
+- Find the link to the loom presentation video here [video](https://www.loom.com/share/317d06bee0724461b76ad07703a441f8)
```
Comment (thecodechaser) [inline README.md]: - [x] Please make sure the maximum length of the presentation video is 5 minutes long and in the video kindly highlight some interesting piece of code as requested in the project requirements. Currently, the video you submitted is 7+ minutes long and it doesn't include highlighted code.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @patriciachrysy👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️ 
- Good job with validations. ✔️ 
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by patriciachrysy** [comment]
The project has been updated according to the remarks, thank you for reviewing again

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @patriciachrysy , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Awesome job, **Patricia**! You have successfully implemented the requested changes. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**